packages/components/input/src/use-input.ts

```diff
@@ -511,7 +511,7 @@ export function useInput<T extends HTMLInputElement | HTMLTextAreaElement = HTML
       return {
         ...props,
         role: "button",
-        tabIndex: 0,
+        tabIndex: originalProps?.isDisabled ? -1 : 0,
```

Comment:
It seems like you can use the button element and pass the `disabled` attribute, which would remove the need for setting `tabindex` and `role: "button"`.
https://github.com/nextui-org/nextui/blob/bed4a59b8a89240cf9706266a31d7e554a627e87/packages/components/input/src/input.tsx#L39

Reply:
Thank you for your comment! 

I agree with using the button element.
Do you think it would be better to proceed with that change?

Reply:
Yes, I think that would be better.